types/apollo-codegen/index.d.ts

```diff
@@ -0,0 +1,40 @@
+// Type definitions for apollo-codegen 0.16
+// Project: https://github.com/apollographql/apollo-codegen
```

Comment:
This project doesn't bundle types, but it is written by Typescript, just adding `"declaration": true` in `tsconfig.json` and `types` in `package.json` can be a simpler way, so you can try to ask the package maintainer whether it is doable.

Reply:
The project is only partially converted to TypeScript (there's lots of `any` types). It's on their roadmap to get to the point where they can publish declarations, but they're not there yet.